runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkBatchTransformTranslators.java

```diff
@@ -276,6 +291,69 @@ public void translateNode(
 
   }
 
+  private static class GBKIntoKeyedWorkItemsTranslator<K, InputT>
```

Comment:
Does this duplicate code with the streaming codepath? If so, can it be deduplicated into a shared location?

Reply:
There is no code duplication since the streaming codepath uses the Flink `DataStream` API while the batch codepath uses the `DataSet` API.

Flink is interesting in that the underlying execution engine is a streaming execution engine but there are two different APIs on top of that. Seems somewhat the opposite of what Dataflow was/is like.

Reply:
Oh okay - then no objections :)